go-packages/meep-sessions/session-mgr.go

```diff
@@ -114,3 +125,56 @@ func (sm *SessionMgr) Authorizer(inner http.Handler) http.Handler {
 		}
 	})
 }
+
+// StartSessionWatchdog -
+func (sm *SessionMgr) StartSessionWatchdog(handler SessionTimeoutHandler) {
+	// Update Watchdog callback function
+	sm.wdHandler = handler
+
+	// Start Session Watchdog ticker to monitor timed out sessions
+	if !sm.wdStarted {
+		sm.wdStarted = true
+		sm.wdTicker = time.NewTicker(wathdogInterval * time.Second)
+		go func() {
+			for range sm.wdTicker.C {
+				if sm.wdStarted {
+					ss := sm.GetSessionStore()
+
+					// Get all sessions
+					sessionList, err := ss.GetAll()
+					if err != nil {
+						log.Warn("Failed to retrieve session list")
+						continue
+					}
+
+					// Remove timed out sessions
+					currentTime := time.Now()
+					for _, session := range sessionList {
+						if currentTime.After(session.Timestamp.Add(SessionDuration * time.Second)) {
+							// Invoke watchdog timeout handler
+							if sm.wdHandler != nil {
+								sm.wdHandler(session)
```

Comment:
- Not invoking sm.wdHandler here could leak a sandbox.
- Is this "if" statement necessary? Since the handler is set a few lines above
- I think it would be safer to make sure the handler is never null & get rid of if statement.

Reply:
Code changes:
- StartSessionWatchdog() now returns an error code. Error cases:
  - Handler is nil
  - Watchdog already runnning
This guarantees that a running Session Watchdog will have a valid handler